cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -3178,6 +3181,13 @@
       (let ((size OperandSize (raw_operand_size_of_type ty)))
         (icmp_cond_result (x64_cmp size b a) cc)))
 
+;; As a special case, reverse the arguments to the comparison when the LHS is a
+;; constant. This ensures that we avoid moving the constant into a register when
+;; performing the comparison.
+(rule 1 (emit_cmp cc (and (simm32_from_value a) (value_type ty)) b)
```

Comment:
I'm not super happy with the need for a rule priority here, and would have expected the `simm32_from_value` use to render it unnecessary. Any idea why it doesn't fire without higher priority?

Reply:
This can go away once #4661 is merged.